src/control/controlframeposproxy.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "audio/frame.h"
+#include "control/controlproxy.h"
+
+/// Special kind of `ControlProxy` that is used for storing frame positions.
+//
+/// Although it exposes a `mixxx::audio::FramePos`-based API, the underlying
+/// control value uses sample positions for backwards compatibility. This might
+/// change in the future.
+class ControlFramePosProxy : public ControlProxy {
+    Q_OBJECT
+  public:
+    /// Creates a new ControlFramePos with the given default position.
+    ControlFramePosProxy(const ConfigKey& key, QObject* pParent = nullptr);
+
+    /// Returns the `mixxx:audio::FramePos` interpretation of the ControlProxy.
+    mixxx::audio::FramePos toFramePos() const {
+        return mixxx::audio::FramePos::fromEngineSamplePosMaybeInvalid(ControlProxy::get());
+    }
+
+    /// Set the ControlProxy from a `mixxx::audio::FramePos`.
+    void set(mixxx::audio::FramePos position) {
```

Comment:
How about adding 2 specializations:
- `setValid(mixxx::audio::FramePos position)` that invokes toEngineSamplePos()
- `setInvalid()` or `unset()` without parameters

In many situations we can decide at compile time which one is desired.

Reply:
We already have that (`reset()`).